lib/matplotlib/axes/_secondary_axes.py

```diff
@@ -86,8 +87,18 @@ def set_location(self, location):
             orientation='y'. A float indicates the relative position on the
             parent axes to put the new axes, 0.0 being the bottom (or left)
             and 1.0 being the top (or right).
+
+        transform : `.Transform`, optional
+            Transform for the location to use. Defaults to
+            the parent's ``transAxes``, so locations are normally relative to
+            the parent axes.
+
+            .. versionadded:: 3.8
         """
 
+        # Make sure transform is a Transform or None.
+        _api.check_isinstance((transforms.Transform, None), transform=transform)
```

Comment:
I wouldn't bother with the comment; the function call is clear:
```suggestion
        _api.check_isinstance((transforms.Transform, None), transform=transform)
```